File: frontend/src/app/services/menu.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Menu } from '../models/Menu';
import { Permission } from '../models/Permission';

@Injectable({
  providedIn: 'root'
})
export class MenuService {

  private apiUrl = 'http://localhost:8080/menus';  

  constructor(private http: HttpClient) {}

  
  getAllMenus(): Observable<Menu[]> {
    return this.http.get<Menu[]>(this.apiUrl);  
  }
  getPermissionsByMenu(idMenu : number): Observable<Permission[]> {
    return this.http.get<Permission[]>(`${this.apiUrl}/getPermissionsByMenu/${idMenu}`);  
  }
}

File: backend/src/main/java/com/pfe/backend/Service/ServiceMenu/MenuService.java
package com.pfe.backend.Service.ServiceMenu;

import com.pfe.backend.Model.Menu;
import com.pfe.backend.Model.Permission;

import java.util.List;

public interface MenuService {
    List<Menu> getAllMenus();
    List<Permission> getPermissionsByMenu(long idMenu);
}
